dataset_interfaces/factory.py

```diff
@@ -60,3 +62,33 @@ def create_examples(dataset_config: dict, universal_args: dict, max_message_size
             if example.example_id == "":
                 example.example_id = str(i)
         return examples
+
+    @staticmethod
+    def create_dataset_for_example(run_configuration: dict,  test_example_path: str) -> DatasetInterface:
+        args = deepcopy(run_configuration["datasets"]["args"])
+
+        # Get the name of the dataset
+
+        path_dataset_name = test_example_path.split(os.sep)[-2]
+        dataset = DATASETS_BY_NAME.get(path_dataset_name, None)
+
+        if dataset is None:
+            raise ValueError(f"No dataset could be resolved from TestExample path: {test_example_path}. Tried {path_dataset_name}")
+
+        # Use the class to get the 'config name'
+        config_name = ""
+        for name, ds in DATASETS.items():
+            if ds == dataset:
+                config_name = name
+                break
```

Comment:
You can also use `DATASETS_BY_NAME`

Reply:
No I can't. `DATASETS_BY_NAME` uses the "nice" name that is contained in `ds.name`. I need to get the name we use in the configuration files, which is only contained in `DATASETS`

Reply:
Ah you're right. I didn't read it thoroughly and thought it was a reverse dict.